roulette/src/main.leo

```diff
@@ -0,0 +1,87 @@
+// The 'roulette' main function.
+record Token {
+    owner: address,
+    gates: u64,
+    amount: u64
+}
+
+function psd_hash(public a: u32) -> field {
+    return Poseidon2::hash(a);
+}
+
+function mint_casino_token_record(a: address, b: u64) -> Token {
+    return Token {
+        owner: a,
+        gates: 0u64,
+        amount: b,
+    };
+}
+
+function make_bet(
+    casino: Token, 
+    player: address, 
+    spin: u8, 
+    bet: u8, 
+    bet_amount: u64,
+    balance: u64,
+) -> (Token, Token) {
+    let win: bool = spin == bet;
+
+    let reward: u64 = bet_amount * 35u64;
+
+    let casino_balance: u64 = 0u64;
+    let player_balance: u64 = 0u64;
+
+    if win {
+        casino_balance = casino.amount - bet_amount;
```

Comment:
When you win, the balances results should be  
`casino_balance = casino.amount - reward`
`player_balance = player.amount + reward`